### PR TITLE
 feat: Remove travis scripts and add github actions for release purpose

### PR DESCRIPTION
**Pull request by flacatus**
Signed-off-by: Flavius Lacatusu <flacatus@redhat.com>

### What does this PR do?
Remove travis scripts and add github actions to release stable and next versions of chectl.

Refference issue:https://github.com/eclipse/che/issues/18260

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by flacatus** [comment]
I'm testing this changes agains my fork and releases looks identical: https://github.com/flacatus/chectl
https://github.com/flacatus/chectl/releases/tag/20201214143842

**Review by tolusha — APPROVED** [review]

File: .github/workflows/release-next.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: Build and Deploy
+on:
+  # Trigger the workflow on push only for the master branch
+  push:
+    branches:
+      - master
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
```
Comment (benoitf) [inline .github/workflows/release-next.yml:9]: Should be a fixed version like Ubuntu 18.04 or 20.04, etc

File: .github/workflows/release-next.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2.3.1 # If you're using actions/checkout@v2 you must set persist-credentials to false in most cases for the deployment to work correctly.
+        with:
+          persist-credentials: false
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '10'
```
Comment (benoitf) [inline .github/workflows/release-next.yml:17]: We can't use 12.x ? The new LTS

File: .github/workflows/release-next.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      - name: Generate tag utilities
+        id: TAG_UTIL
+        run: |
+          echo "::set-output name=short_sha::$(git rev-parse --short HEAD)"
+          echo "::set-output name=current_hour::$(date +'%Y%m%d%H%M%S')"
+          echo "::set-output name=current_date::$(date +'%Y%m%d')"
+      - name: Install and Build 🔧 # This example project is built using npm and outputs the result to the 'build' folder. Replace with the commands required to build your project, or remove this step entirely if your site is pre-built.
+        run: |
+          git config --local user.name "Flavius Lacatusu"
```
Comment (benoitf) [inline .github/workflows/release-next.yml:26]: I think we should use there the author that merge to master

Comment (flacatus) [reply .github/workflows/release-next.yml]: Right!

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -1,29 +1,79 @@
...
+            SHORT_SHA1=$(git rev-parse --short HEAD)
+            export CHECTL_VERSION=0.0.$CURRENT_DAY-next
+            echo "::set-output name=chectl_version::0.0.$CURRENT_DAY-next"
+            echo "::set-output name=gh_release_name::0.0.$CURRENT_DAY-next.$SHORT_SHA1"
+          fi
+      - name: Install chectl dependencies 🔧
+        run: yarn
+      - name: Run unit tests 🩺
+        run: yarn test
+      - name: Pack chectl 🎒
```
Comment (nickboldt) [inline .github/workflows/release.yml:50]: I kinda hate these emojis, as I expect about half of them won't work correctly if I paste console output into a JIRA... :(

Comment (flacatus) [reply .github/workflows/release.yml:50]: I'll remove them

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -1,29 +1,79 @@
...
+      - name: Generate tag utilities 📜
+        id: TAG_UTIL
         run: |
-          # since we're using Flavius' token below to generate PRs, should also make tags/branches/commits by same owner
-          # see https://github.com/che-incubator/chectl/settings/secrets/actions for the token used
-          git config --global user.name "Flavius Lacatusu"
-          git config --global user.email "flacatus@redhat.com"
-          export GITHUB_TOKEN=${{ secrets.ACTIONS_TOKEN }}
-          /bin/bash -x make-release.sh --version ${{ github.event.inputs.version }}
+          echo "::set-output name=gh_tag::$(date +'%Y%m%d%H%M%S')"
```
Comment (nickboldt) [inline .github/workflows/release.yml:32]: what do these `::set-output ...` echos do? are they just informative console info?

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -1,29 +1,79 @@
...
+          then
+            echo "::set-output name=chectl_version::$(cat VERSION)" 
+            echo "::set-output name=gh_release_name::$(cat VERSION)"
+          fi
+          if [[ ${GITHUB_REF#refs/heads/} =~ master ]];
+          then
+            CURRENT_DAY=$(date +'%Y%m%d')
+            SHORT_SHA1=$(git rev-parse --short HEAD)
+            export CHECTL_VERSION=0.0.$CURRENT_DAY-next
+            echo "::set-output name=chectl_version::0.0.$CURRENT_DAY-next"
```
Comment (nickboldt) [inline .github/workflows/release.yml:43]: should these echos be piped to a file?

File: .github/workflows/release.yml (hunk 5 above)
Comment (flacatus) [reply .github/workflows/release.yml:32]: ::set-output I'm setting the necessary tags and then I can use it in all part of job. Works like envs

File: .github/workflows/unit-tests.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+#    This program and the accompanying materials are made
+#    available under the terms of the Eclipse Public License 2.0
+#    which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+#  SPDX-License-Identifier: EPL-2.0
+#
+#  Contributors:
+#    Red Hat, Inc. - initial API and implementation
+name: Chectl
+on: [pull_request, push]
```
Comment (nickboldt) [inline .github/workflows/unit-tests.yml:12]: if we enforce that we must submit a PR for every change, do we need to run tests on every push too?

Comment (flacatus) [reply .github/workflows/unit-tests.yml:12]: oh, my mistake... Only in PRs

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -1,29 +1,79 @@
...
-    runs-on: ubuntu-latest
+  build-and-deploy:
+    runs-on: ubuntu-20.04
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout 🛎️
+        uses: actions/checkout@v2.3.1
         with:
-          fetch-depth: 0
-          token: ${{ secrets.ACTIONS_TOKEN }}
```
Comment (nickboldt) [inline .github/workflows/release.yml:18]: why don't you need to fetch code using the token?

File: .github/workflows/release.yml (hunk 4 above)
Comment (nickboldt) [reply .github/workflows/release.yml:50]: just personal preference, I guess ;) most of the chectl emojis DO work in a paste, but a couple don't... which causes weird problems reporting jiras.

File: .github/workflows/release.yml (hunk 8 above)
Comment (flacatus) [reply .github/workflows/release.yml:18]: Good question. I didn't change nothing. Comes from here: https://github.com/che-incubator/chectl/blob/remove_travis/.github/workflows/draft-release.yml#L14-L18

File: .github/workflows/release.yml (hunk 5 above)
Comment (nickboldt) [reply .github/workflows/release.yml:32]: so `TAG_UTIL` somehow gets the variables ?

File: .github/workflows/release.yml (hunk 8 above)
Comment (nickboldt) [reply .github/workflows/release.yml:18]: ok

File: .github/workflows/release.yml (hunk 5 above)
Comment (flacatus) [reply .github/workflows/release.yml:32]: Yeah like here.https://github.com/che-incubator/chectl/blob/remove_travis/.github/workflows/release.yml#L56
I found the docs today here: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#using-workflow-commands-to-access-toolkit-functions

Comment (nickboldt) [reply .github/workflows/release.yml:32]: well, that's an odd syntax, but hey, if that's the best way to have global vars persist between steps, so be it.

**Review by nickboldt — APPROVED** [review]
LGTM but I didn't run this. Deferring to others for final say. +0.1

**Comment by flacatus** [comment]
@nickboldt https://github.com/flacatus/chectl/releases
You can try in my fork

File: .github/workflows/release.yml (hunk 5 above)
Comment (benoitf) [reply .github/workflows/release.yml:32]: Just a FYI you can use syntax like to share env variables ( and not step output)

echo "PR_NUMBER=$pr_number" >> $GITHUB_ENV

File: .github/workflows/draft-release.yml [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+          fetch-depth: 0
+          token: ${{ secrets.ACTIONS_TOKEN }}
+      - uses: actions/setup-node@v2.1.2
+        with:
+          node-version: 12.18.2
+      - name: Run make-release.sh script
+        run: |
+          # since we're using Flavius' token below to generate PRs, should also make tags/branches/commits by same owner
+          # see https://github.com/che-incubator/chectl/settings/secrets/actions for the token used
+          git config --global user.name "Flavius Lacatusu"
```
Comment (benoitf) [inline .github/workflows/draft-release.yml:26]: Should it be author that perform the release ?

File: .github/workflows/release-next.yml (hunk 3 above)
Comment (flacatus) [reply .github/workflows/release-next.yml]: @benoitf How can I get the author in GH actions?

Comment (flacatus) [reply .github/workflows/release-next.yml]: {{ github.actor }} ?

**Review by benoitf — APPROVED** [review]
let's try

**Pull request merged**